emr-datahub/src/main/scala/org/apache/spark/sql/aliyun/datahub/DatahubOffsetReader.scala

```diff
@@ -227,15 +232,28 @@ object DatahubOffsetReader extends Logging with Serializable {
       throw new MissingArgumentException("Missing access key secret (='access.key.secret')."))
     val endpoint = sourceOptions.getOrElse("endpoint",
       throw new MissingArgumentException("Missing endpoint (='endpoint')."))
-    if (DatahubClient == null) {
-      DatahubClient = new DatahubClientAgent(new DatahubConfiguration(new AliyunAccount(accessKeyId, accessKeySecret),
-        endpoint))
-    }
-    DatahubClient
+    getOrCreateDatahubClient(accessKeyId, accessKeySecret, endpoint)
   }
 
   def resetConsumer(sourceOptions: Map[String, String]): Unit = synchronized {
```

Comment:
will cause multiple thread frequently remove and recreate consumer when consumer is error